common/app/implicits/Requests.scala

```diff
@@ -45,6 +45,8 @@ trait Requests {
     lazy val campaignCode: Option[String] = r.getQueryString("CMP")
 
     lazy val isAdFree: Boolean = r.headers.keys.exists(_ equalsIgnoreCase "X-Gu-Commercial-Ad-Free")
+
+    lazy val referrer: Option[String] = r.headers.get("referer")
```

Comment:
is the header name `referer` or `referrer`?

Reply:
Madly, it's called `referer`.
https://en.wikipedia.org/wiki/HTTP_referer#cite_note-1

Reply:
😂